src/Elliptic/Protocols/FirstOrderSystem.hpp

```diff
@@ -0,0 +1,207 @@
+ *
+ * \f{equation}
+ * -\partial_i F^i_\alpha + S_\alpha = f_\alpha(x)
+ * \f}
+ *
+ * in terms of fluxes \f$F_\alpha^i\f$, sources \f$S_\alpha\f$ and fixed-sources
+ * \f$f_\alpha(x)\f$. It resembles closely formulations of hyperbolic
+ * conservation laws but allows the fluxes \f$F_\alpha^i\f$ to be higher-rank
+ * tensor fields. The fluxes and sources are functionals of the "primal" system
+ * variables \f$u_A(x)\f$ and their corresponding "auxiliary" variables
+ * \f$v_A(x)\f$. The fixed-sourced \f$f_\alpha(x)\f$ are independent of the
+ * system variables. We enumerate the variables with uppercase letters such that
+ * \f$v_A\f$ is the auxiliary variable corresponding to \f$u_A\f$. Greek letters
+ * enumerate _all_ variables. In documentation related to particular elliptic
+ * systems we generally use the canonical system-specific symbols for the fields
+ * in place of these indices. See the `Poisson::FirstOrderSystem` and the
+ * `Elasticity::FirstOrderSystem` for examples.
+ *
+ * Conforming classes must have these static member variables:
+ *
+ * - `size_t volume_dim`: The number of spatial dimensions.
+ *
+ * Conforming classes must have these type aliases:
+ *
+ * - `primal_fields`: A list of tags representing the primal fields. These are
+ *   the fields we solve for, e.g. \f$u\f$ for a Poisson equation.
+ *
+ * - `auxiliary_fields`: A list of tags representing the auxiliary fields, which
+ *   are typically gradients of the primal fields, e.g. \f$v_i = \partial_i u\f$
+ *   for a Poisson equation. These must follow the order of the `primal_fields`.
+ *   Specifically, each auxiliary field must have one rank higher than its
+ *   corresponding primal field.
+ *
+ * - `primal_fluxes`: A list of tags representing the primal fluxes
+ *   \f$F_{u_A}^i\f$. These are typically some linear combination of the
+ *   auxiliary fields with raised indices, e.g. \f$v^i = g^{ij}v_j\f$ for a
+ *   curved-space Poisson equation. They must have the same rank as the
+ *   `auxiliary_fields` but with an upper-spatial first index, because their
+ *   divergence defines the elliptic equation.
+ *
+ * - `auxiliary_fluxes`: A list of tags representing the auxiliary fluxes
+ *   \f$F_{v_A}^i\f$, e.g. \f$\delta^i_j u\f$ for a Poisson equation. These must
+ *   have one rank higher than the `auxiliary_fields` and have an upper-spatial
+ *   first index because their divergence defines the auxiliary fields.
+ *
+ * - `background_fields`: A list of tags representing the variable-independent
+ *   background fields in the equations. Examples are a background metric,
+ *   associated fixed geometry quantities such as Christoffel symbols or the
+ *   Ricci scalar, or any other fixed field that determines the problem to be
+ *   solved such as matter sources in the Einstein constraint equations.
+ *
+ * - `inv_metric_tag`: The tag that defines the background geometry. This is the
+ *   metric responsible for normalizing one-forms, such as face normals.
```

Comment:
can you clarify if this should be the spatial metric or the spatial part of the spacetime metric?

Reply:
It's up to the system what this metric is :) For a curved-space Poisson eqn it's just the background metric. For the XCTS equations it's the conformal metric. I think I can't really make it more specific at this point than "the geometry that the equations are formulated on".

Reply:
Is it not the spatial metric of the background geometry? I'm asking whether it is a spatial metric, spatial components of a spacetime metric, or the spacetime metric. At least in the evolution code this decision is made regardless of what the background geometry is (we use the inverse spatial metric)

Reply:
There is only one metric in the elliptic equations, and that's the one coupling the principal part of the PDEs. It's just "the background metric g_ij" for a curved-space Poisson equation, and it's "the conformal metric \bar{gamma}_ij" for the XCTS equations. For the XCTS equations the conformal metric is used to assemble the spacetime metric, by combining it with the conformal factor, lapse, shift etc.

Reply:
okay, thanks :)